packages/mui-material/src/TablePagination/TablePaginationActions.js

```diff
@@ -48,39 +50,41 @@ const TablePaginationActions = React.forwardRef(function TablePaginationActions(
       {showFirstButton && (
         <IconButton
           onClick={handleFirstPageButtonClick}
-          disabled={page === 0}
+          disabled={disabled || page === 0}
           aria-label={getItemAriaLabel('first', page)}
           title={getItemAriaLabel('first', page)}
+          {...(slotProps?.firstButton ?? {})}
         >
           {theme.direction === 'rtl' ? <LastPageIcon /> : <FirstPageIcon />}
         </IconButton>
       )}
       <IconButton
         onClick={handleBackButtonClick}
-        disabled={page === 0}
+        disabled={disabled || page === 0}
         color="inherit"
         aria-label={getItemAriaLabel('previous', page)}
         title={getItemAriaLabel('previous', page)}
-        {...backIconButtonProps}
+        {...(slotProps?.previousButton ?? backIconButtonProps)}
```

Comment:
Do we have tests for the `backIconButtonProps` and `nextIconButtonProps` to ensure we're not breaking other developer's experience with this change?

Reply:
I've added some tests in https://github.com/mui/material-ui/pull/39353/commits/4dcf345e607c917460d1f897d4c16aa1b3dcb12a. Lmk what you think!

Reply:
Nice 👍🏼